### PR TITLE
Complete new UI design for self registration and account recovery

### PR DESCRIPTION
**Pull request by vihanga-liyanage**
## Purpose
- Added the first half of the UI changes in server configs.

image

Refers https://github.com/wso2/identity-apps/issues/772

### CODE REVIEW

File: modules/theme/src/themes/default/modules/accordion.overrides [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+             .chevron {
+                 transform: rotate(90deg);
+             }
+         }
+    }
+}
+
+.ui.accordion {
+    .title {
+        .disabled {
```
Comment (brionmario) [inline modules/theme/src/themes/default/modules/accordion.overrides:20]: This style will be applied to the elements with `.disabled` classname in all the accordions.
Shall we scope this to the specific accordion?

Comment (vihanga-liyanage) [reply modules/theme/src/themes/default/modules/accordion.overrides]: I couldn't find a prop in React Accordion to disable. Can't we use this for all accordions?

File: apps/admin-portal/src/components/server-configurations/user-self-registration.tsx [hunk 2]
```diff
@@ -278,41 +261,61 @@ export const UserSelfRegistration: FunctionComponent<UserSelfRegistrationProps>
...
+							Internal notification management
+						</label>
+					</Grid.Column>
+					<Grid.Column mobile={ 2 } tablet={ 2 } computer={ 2 }>
+                        <Field
+                            name={ ServerConfigurationsConstants.SELF_SIGN_UP_NOTIFICATIONS_INTERNALLY_MANAGED }
+                            required={ false }
+                            requiredErrorMessage=""
+                            type="checkbox"
+                            children={ [
```
Comment (DimalChandrasiri) [inline apps/admin-portal/src/components/server-configurations/user-self-registration.tsx:282]: Unwanted indentation

File: apps/admin-portal/src/components/server-configurations/account-recovery.tsx [hunk 3]
```diff
@@ -156,57 +126,44 @@ export const AccountRecovery: FunctionComponent<AccountRecoveryProps> = (props:
 		makeAccountRecoveryPatchCall(data, successNotification);
 	};
 
-	const saveAccountRecoveryAdvancedConfigs = (accountRecoveryConfigs) => {
+	const saveAccountRecoveryAdvancedConfigs = (configs) => {
 		const data = {
```
Comment (DimalChandrasiri) [inline apps/admin-portal/src/components/server-configurations/account-recovery.tsx:130]: IMO the indentation is more than 4 spaces which is the standard used. Could we look into this?

File: apps/admin-portal/src/components/server-configurations/account-recovery.tsx [hunk 4]
```diff
@@ -50,672 +44,908 @@ interface AccountRecoveryProps {
...
-					{ showAccountRecoverySummary }
-					{ showUserAccountRecoveryView }
-				</List.Item>
-			</List>
-		</Section>
-	);
+    const [ accountRecoveryConfigs, setAccountRecoveryConfigs ] = useState<AccountRecoveryConfigurationsInterface>({});
+    const [ mainAccordionActiveState, setMainAccordionActiveState ] = useState<boolean>(false);
+    const [ subAccordionActiveIndex, setSubAccordionActiveIndex ] = useState<number>(undefined);
+    const [ reset] = useTrigger();
```
Comment (brionmario) [inline apps/admin-portal/src/components/server-configurations/account-recovery.tsx:50]:
```suggestion
    const [ reset ] = useTrigger();
```

**Pull request merged**